registry/registry.go

```diff
@@ -89,9 +89,25 @@ func WrapTransport(transport http.RoundTripper, url, username, password string)
 	return errorTransport
 }
 
-func newFromTransport(registryUrl, username, password string, transport http.RoundTripper, logf LogfCallback) (*Registry, error) {
+func WrapTokenTransport(transport http.RoundTripper, url, username, password string) http.RoundTripper {
```

Comment:
Could you please add a comment to document how the `RoundTripper` returned from `WrapTokenTransport` behaves differently from the one returned by `WrapTransport`, so that consumers will be able to tell which one they want?

Reply:
Also, it seems like `WrapTransport` and `WrapTokenTransport` could be de-duped, perhaps:

```go
func WrapTransport(transport http.RoundTripper, url, username, password string) http.RoundTripper {
	return wrapTransport(transport, url, username, password, true, true)
}

func WrapTokenOnlyTransport(transport http.RoundTripper, url, username, password string) http.RoundTripper {
	return wrapTransport(transport, url, username, password, true, false)
}

func wrapTransport(transport http.RoundTripper, url, username, password string, withTokenAuth, withBasicAuth bool) http.RoundTripper {
	if withTokenAuth {
		transport = &TokenTransport{
			Transport: transport,
			Username: username,
			Password: password,
		}
	}

	if withBasicAuth {
		transport = &BasicTransport{
			Transport: transport,
			URL: url,
			Username: username,
			Password: password,
		}
	}

	transport := &ErrorTransport{
		Transport: transport,
	}

	return transport
}
```

Thoughts?